build-support/bin/release.sh

```diff
@@ -42,11 +94,16 @@ function requirement() {
   grep "^${package}[^A-Za-z0-9]" "${ROOT}/3rdparty/python/requirements.txt" || die "Could not find requirement for ${package}"
 }
 
-function run_pex27() {
+function run_pex() {
   # TODO: Cache this in case we run pex multiple times
   (
     PEX_VERSION="$(requirement pex | sed -e "s|pex==||")"
     PEX_PEX=pex27
+    if [[ "${python_three}" == "true" ]]; then
```

Comment:
I think it does make a difference.

For example, with all the other changes made in this script but removing this section of code to always use pex27 no matter what, `./build-support/bin/release.sh -3n` will fail with this error:

```python
Traceback (most recent call last):
  File "/private/var/folders/sx/pdpbqz4x5cscn9hhfpbsbqvm0000gn/T/tmpEMUnm5/.bootstrap/_pex/pex.py", line 349, in execute
    exit_code = self._wrap_coverage(self._wrap_profiling, self._execute)
  File "/private/var/folders/sx/pdpbqz4x5cscn9hhfpbsbqvm0000gn/T/tmpEMUnm5/.bootstrap/_pex/pex.py", line 281, in _wrap_coverage
    return runner(*args)
  File "/private/var/folders/sx/pdpbqz4x5cscn9hhfpbsbqvm0000gn/T/tmpEMUnm5/.bootstrap/_pex/pex.py", line 312, in _wrap_profiling
    return runner(*args)
  File "/private/var/folders/sx/pdpbqz4x5cscn9hhfpbsbqvm0000gn/T/tmpEMUnm5/.bootstrap/_pex/pex.py", line 397, in _execute
    return self.execute_interpreter()
  File "/private/var/folders/sx/pdpbqz4x5cscn9hhfpbsbqvm0000gn/T/tmpEMUnm5/.bootstrap/_pex/pex.py", line 456, in execute_interpreter
    self.execute_content(arg, content)
  File "/private/var/folders/sx/pdpbqz4x5cscn9hhfpbsbqvm0000gn/T/tmpEMUnm5/.bootstrap/_pex/pex.py", line 491, in execute_content
    exec_function(ast, globals())
  File "<exec_function>", line 4, in exec_function
  File "/Users/eric/DocsLocal/code/projects/pants/src/python/pants/releases/packages.py", line 13, in <module>
    from configparser import ConfigParser
ImportError: No module named configparser
```

While we can get around that by passing the backport as a requirement always, whereas now we only pass if releasing with Py2, this failure speaks to the fact that `pex27` indeed means most the processes will be run using Py2.

--

I think we do want to execute every process with Py3 when `-3` is enabled. For the same reason John agreed over Slack we should require 3.6 instead of 3.7 when building Py3 wheels (like we use `xcode8` and Docker travis), we want to use Py3.6 to build the wheel for max compatibility.